### PR TITLE
Add support for G6+ sensor rx message changes

### PR DESCRIPTION
**Pull request by jpcunningh**

### CODE REVIEW

File: lib/transmitter.js [hunk 1]
```diff
@@ -285,7 +286,10 @@ async function readGlucose() {
 async function readSensorMessage() {
   try {
     const message = new SensorTxMessage();
-    const data = await this.manager.writeValueAndWaitForNotification(message.data, uuid.Control);
+    let data = await this.manager.writeValueAndWaitForNotification(message.data, uuid.Control);
+    if (data[0] === D0UnknownRxMessage.opcode) {
+      data = await this.manager.waitForNotification(uuid.Control);
```
Comment (efidoman) [inline lib/transmitter.js]: Is this what you were referring to? I'm wondering why we have to effectively wait twice in the case of opcode D0UnknownRxMessage?

File: lib/transmitter.js [hunk 2]
```diff
@@ -323,6 +327,7 @@ module.exports = class Transmitter extends EventEmitter {
 
     // if serial number starts with 8xxxxx then assume g6
     this.g6Transmitter = (id.substr(0, 1) === '8');
+    this.g6PlusTransmitter = (id.substr(0, 2) === '8G');
```
Comment (efidoman) [inline lib/transmitter.js]: I suppose we can assume all g6 new firmware transmitters have a serial number that starts with "8G" until / if someone reports differently.

**Comment by jpcunningh** [comment]
I believe so for now. I’m going to try to implement retrieving the firmware
version this week to give us a better option for determining which
interface version we need to use.

File: lib/transmitter.js (hunk 2 above)
Comment (efidoman) [reply lib/transmitter.js]: Are you going to use the new version request value here now? Is that possible?

**Comment by jpcunningh** [comment]
Do you think we should? It’s more complicated than just using the ID, and I
don’t know if a case yet where we have a difference we have to deal with
that can’t be determined based on ID.

I was planning on leaving it as is using the ID until we need to use the
Firmware Version to keep it as simple as possible until we had to do
something else.  I wanted to get the firmware version working so when that
need arises, we can shorten the time it takes to address it rather than
having to implement the code to obtain the version information first. That
also gives us some more time to think through how to best implement the
firmware version based decisions.

Jeremy

**Comment by efidoman** [comment]
That works for me. I will add  logger command line utility to optionally send a version request to xdrip-js so I can test that.

**Comment by efidoman** [comment]
I tested with Logger and it works great. Ready to Merge.

2019-10-01T15:32:31.726Z bluetooth-manager Tx 206224
2019-10-01T15:32:31.746Z bluetooth-manager Rx 210001060d8b0101002700064959f3009a79
2019-10-01T15:32:31.750Z bluetooth-manager Tx 4a8ee9
2019-10-01T15:32:31.776Z bluetooth-manager Rx 4b00010605199b2b00000000036e00f0001bc6
2019-10-01T15:32:31.779Z bluetooth-manager Tx 52b77a
2019-10-01T15:32:31.806Z bluetooth-manager Rx 53000a07005f5f5f5f524c53455f5f5f5f602b
2019-10-01T15:32:31.808Z transmitter got version messages: 1.6.5.25

**Pull request merged**